### PR TITLE
Working auth

### PR DESCRIPTION
**Pull request by tvaroglu**
- Changes Implemented:
  - Updated `BackEndService` to call new user auth endpoint from BE, to allow for current_user helper method call to BE in our `helper_method`
  - Added placeholder in `BackEndService.base_url` method call, to enable testing via local host

- Quality Control Checklist:

  - [x] Code adheres to Rubocop styling (if unavoidable infractions, please clarify below)
    - We have some work to do here - will have to come back to some items
  - [x] 100% SimpleCov test coverage (if below, please clarify below)
    - Stubbed methods still remain untested
  - [x] Last commit has passing Travis build

- Blockers (if applicable):

- Next Steps & Additional Notes:

### CODE REVIEW

File: app/facades/back_end_facade.rb [hunk 1]
```diff
@@ -11,23 +11,17 @@ def create_user(params)
 
     def get_user_friends(user_id)
       friends = BackEndService.get_friends(user_id)
-      friends[:data].map do |friend|
-        User.new(friend[:attributes])
-      end
+      friends[:data].map { |friend| User.new(friend[:attributes]) } if friends[:data]
```
Comment (Scott-Borecki) [inline app/facades/back_end_facade.rb:14]: Nice!  I like the refactor here!

**Review by gunnarrunner — APPROVED** [review]
Awesome work!!! Looks good.

File: app/services/back_end_service.rb [hunk 2]
```diff
@@ -11,7 +11,7 @@ def get_friends(user_id)
     end
 
     def get_gyms(user_id)
-      response = db_conn.get("/api/v1/users/#{user_id}/gyms")
+      response = db_conn.get("/api/v1/users/#{user_id}/gym_memberships")
```
Comment (Scott-Borecki) [inline app/services/back_end_service.rb:14]: Nice!  Thanks for updating this!

**Review by Scott-Borecki — APPROVED** [review]
Looks good!  Thanks for patching that together @tvaroglu !

**Review by ealwafai — APPROVED** [review]
Looks great!

File: app/facades/back_end_facade.rb (hunk 1 above)
Comment (tvaroglu) [reply app/facades/back_end_facade.rb:14]: :D 

I wish Rubocop agreed

**Review by carolinectan — APPROVED** [review]

**Pull request merged**